Packs/GenericAPIEventCollector/Integrations/GenericAPIEventCollector/GenericAPIEventCollector.py

```diff
@@ -0,0 +1,664 @@
+        last_fetched_datetime = datetime.fromisoformat(last_fetched_datetime_str)
+        first_fetch_for_this_integration = False
+    # endregion
+    # region load request arguments
+    request_data: dict[Any, Any] | None = parse_json_param(params.get('request_data'), 'request_data')
+    request_json: dict[Any, Any] | None = parse_json_param(params.get('request_json'), 'request_json')
+    query_params: dict[Any, Any] | None = parse_json_param(params.get('query_params'), 'query_params')
+    pagination_logic = extract_pagination_params(params)
+    # If we've an initial query argument, we try with it.
+    if first_fetch_for_this_integration:
+        if params.get('initial_query_params'):
+            query_params = parse_json_param(params.get('initial_query_params'), 'initial_query_params')
+        if params.get('initial_pagination_params'):
+            pagination_logic = extract_pagination_params(params.get('initial_pagination_params'))
+        if params.get('initial_request_data'):
+            request_data = parse_json_param(params.get('initial_request_data'), 'initial_request_data')
+        if params.get('initial_request_json'):
+            request_json = parse_json_param(params.get('initial_request_json'), 'initial_request_json')
+        if params.get('initial_query_params'):
+            query_params = parse_json_param(params.get('initial_query_params'), 'initial_query_params')
+
+    # endregion
+    # region Handle substitutions
+    substitutions: list[tuple[str, str]] = [
+        (place_holder.value, last_run.get(place_holder.value)) for place_holder in PlaceHolders  # type: ignore[misc]
+        if last_run.get(place_holder.value) is not None
+    ]
+    substitutions_query_params: dict[Any, Any] | None = recursive_replace(query_params, substitutions)
+    substitutions_request_json: dict[Any, Any] | None = recursive_replace(request_json, substitutions)
+    substitutions_request_data: dict[Any, Any] | None = recursive_replace(request_data, substitutions)
+    # endregion
+    return last_fetched_datetime, pagination_logic, RequestData(substitutions_request_data, substitutions_request_json,
+                                                                substitutions_query_params)
+
+
+def iso8601_to_datetime_str(iso8601_time: str) -> str:
+    # In case the time format is ISO 8601 - ISO supports 7 digits while datetime in python supports only 6,
+    # so we need to reduce 1 number from the nanoseconds
+    if '.' in iso8601_time:
+        timestamp_without_nanoseconds, nanoseconds = re.split("[.]", iso8601_time, maxsplit=1)
+        fractional = nanoseconds.rstrip('Z')[:6]  # Keep only the first 6 digits.
+        iso8601_time = f"{timestamp_without_nanoseconds}.{fractional}Z"
+    return iso8601_time
+
+
+def test_module(client: Client,
+                endpoint: str,
+                http_method: str,
+                request_data: RequestData):
+    try:
+        events = client.search_events(endpoint=endpoint,
```

Comment:
When not sending the time, fetch can tak a long time, worth checking for limitation using of starting time of 1 day at most.